tests/test_target_database.py

```diff
@@ -1,46 +1,51 @@
 """Tests for `mbed_targets`."""
 
-import unittest
+from unittest import TestCase
+
 import requests_mock
 
 # Unit under test
 import mbed_targets._internal.target_database as target_database
 
 
-class TestGetTargetData(unittest.TestCase):
+class TestGetTargetData(TestCase):
     """Tests for the method `target_database.get_target_data`."""
 
     @requests_mock.mock()
-    def test_404(self, mock_request):
-        """Given a 404 response, target data is not set."""
-        mock_request.get(target_database._TARGET_API, status_code=404, text="Not Found")
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
+    def test_401(self, mock_request):
+        """Given a 401 error code, TargetAPIError is raised."""
+        mock_request.get(
+            target_database._TARGET_API, status_code=401
+        )
+        with self.assertRaises(target_database.TargetAPIError):
+            target_database.get_target_data()
 
     @requests_mock.mock()
-    def test_401(self, mock_request):
-        """Given a 401 response, target data is not set."""
-        mock_request.get(target_database._TARGET_API, status_code=401)
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
+    def test_404(self, mock_request):
+        """Given a 404 error code, TargetAPIError is raised."""
+        mock_request.get(
+            target_database._TARGET_API, status_code=404, text="Not Found"
+        )
+        with self.assertRaises(target_database.TargetAPIError):
+            target_database.get_target_data()
 
     @requests_mock.mock()
     def test_200_invalid_json(self, mock_request):
-        """Given a valid response but invalid json, target data is not set."""
-        mock_request.get(target_database._TARGET_API, text="invalid_json")
-        target_data = target_database.get_target_data()
-        self.assertIsNone(target_data, "Target data is expected to be empty on an error")
+        """Given a valid response but invalid json, JSONDecodeError is raised."""
+        mock_request.get(target_database._TARGET_API, json=None)
```

Comment:
Is `json=None` actually invalid? I thought this was quite clear and guaranteed to be invalid JSON.